packages/dataclass-editor/src/data/dataclass-hooks.ts

```diff
@@ -0,0 +1,48 @@
+import { useAppContext } from '../context/AppContext';
+import { useEntityField, useField } from '../context/FieldContext';
+import { type DataClass, type DataClassField, type DataClassFieldEntity, type DataClassFieldModifier } from './dataclass';
+
+export const useDataClassProperty = () => {
+  const { dataClass, setDataClass } = useAppContext();
+  const setProperty = <DKey extends keyof DataClass>(key: DKey, value: DataClass[DKey]) => {
+    const newDataClass = structuredClone(dataClass);
+    newDataClass[key] = value;
+    setDataClass(newDataClass);
+  };
+  return setProperty;
+};
+
+export const useFieldProperty = () => {
+  const { field, setField } = useField();
+  const setProperty = <FKey extends keyof DataClassField>(key: FKey, value: DataClassField[FKey]) => {
+    const newField = structuredClone(field);
+    newField[key] = value;
+    setField(newField);
+  };
+  return setProperty;
+};
```

Comment:
same with this, only used in detail/field. why not move there as single file `useFieldProperty.ts`?